File: api-backend/src/inmemory/inmemory.ts
type User = {
  Id: string,
  username: string,
  password: string,
  balance: {
    coins: Record<string, number>,
    usd: number
  }
}
export type Trade = {
  stopLoss?: number | null,
  status: "open" | "closed",
  tradeId: string,
  takeProfit?: number | null,
  pnl?: number,
  openPrice: number,
  leverage: number,
  quantity: number,
  userId: string,
  asset: string,
  closedPrice?: number,
  marginPrice: number
}
export const trades: Trade[] = []
export const liquidity = new Map<string, Array<Trade>>()
export const users: User[] = []

File: engine-backend/src/Inmemory/Inmemory.rs
use rust_decimal::Decimal;
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct User {
    pub username: String,
    pub password: String,
    pub balances: BalanceType,
    pub Id: String,
}
#[derive(Debug, Serialize, Deserialize)]
pub struct BalanceType {
    pub amt: i64,
    pub coin: Vec<CoinType>,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct CoinType {
    pub assert: String,
    pub quantity: Decimal,
}
#[derive(Debug, Deserialize, Serialize)]
pub struct Trades {
    pub tradeId: String,
    pub Type:String,
    pub userId: String,
    pub asset: String,
    pub status: String,
    pub openPrice: Decimal,
    pub leverage: Option<Decimal>,
    pub quantity: Decimal,
    pub marginPrice: Option<Decimal>,
    pub stopLoss: Option<Decimal>,
    pub takeProfit: Option<Decimal>,
}
